sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/tests/TimeSeriesIdTests.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Azure.Iot.TimeSeriesInsights.Tests
+{
+    public class TimeSeriesIdTests : E2eTestBase
+    {
+        private static readonly TimeSpan s_retryDelay = TimeSpan.FromSeconds(10);
+
+        // This is the GUID that TSI uses to represent the default type for a Time Series Instance.
+        // TODO: replace hardcoding the Type GUID when the Types resource has been implemented.
+        private const string DefaultType = "1be09af9-f089-4d6b-9f0b-48018b5f7393";
+        private const int MaxNumberOfRetries = 10;
+
+        public TimeSeriesIdTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [Test]
+        public async Task TimeSeriesInsightsInstances_TimeSeries3Id()
+        {
+            // Arrange
+            TimeSeriesInsightsClient client = GetClient();
+
+            // Create a Time Series Id with 3 properties. Middle property is a null
+            TimeSeriesId idWithNull = new TimeSeriesId(
+                Recording.GenerateAlphaNumericId(string.Empty, 5),
+                null,
+                Recording.GenerateAlphaNumericId(string.Empty, 5));
+
+            var timeSeriesInstances = new List<TimeSeriesInstance>
+            {
+                new TimeSeriesInstance(idWithNull, DefaultType),
+            };
+
+            // Act and assert
+            try
+            {
+                // Create TSI instances
+                Response<InstancesOperationError[]> createInstancesResult = await client
+                    .CreateOrReplaceTimeSeriesInstancesAsync(timeSeriesInstances)
+                    .ConfigureAwait(false);
+
+                // Assert that the result error array does not contain any object that is set
+                createInstancesResult.Value.All((errorResult) => errorResult == null)
```

Comment:
I think we can do: `createInstanceResult.Value.Should().OnlyContain(x => x  == null);`

There are so many interesting overloads and I'm trying to learn more of them to have more natural assertions.

<https://fluentassertions.com/collections/>

Reply:
I like it!